entity-service/src/main/kotlin/com/egm/stellio/entity/authorization/Neo4jAuthorizationRepository.kt

```diff
@@ -10,33 +9,31 @@ class Neo4jAuthorizationRepository(
     private val session: Session
 ) {
 
-    data class AvailableRightsForEntity(
-        val targetEntityId: String,
-        val rights: List<String> = emptyList()
-    )
-
-    fun getAvailableRightsForEntities(userId: String, entitiesId: List<String>): List<AvailableRightsForEntity> {
+    fun filterEntitiesUserHasOneOfGivenRights(
+        userId: String,
+        entitiesId: List<String>,
+        rights: Set<String>
+    ): List<String> {
         val query = """
             MATCH (userEntity:Entity {id : ${'$'}userId}), (entity:Entity)
             WHERE entity.id IN ${'$'}entitiesId
             OPTIONAL MATCH (userEntity)-[:HAS_OBJECT]->(right:Attribute:Relationship)-[]->(entity:Entity)
             OPTIONAL MATCH (userEntity)-[:HAS_OBJECT]->(:Attribute:Relationship)-[:IS_MEMBER_OF]->(:Entity)-[:HAS_OBJECT]->(grpRight:Attribute:Relationship)-[]->(entity:Entity)
-            RETURN entity.id as id, right, grpRight
+            WHERE size([label IN labels(right) WHERE label IN ${'$'}rights | 1]) > 0 AND
+                    size([label IN labels(grpRight) WHERE label IN ${'$'}rights | 1]) > 0
+            RETURN entity.id as id, count(right) AS nbOfRights, count(grpRight) AS nbOfGrpRights
         """.trimIndent()
 
         val parameters = mapOf(
             "userId" to userId,
-            "entitiesId" to entitiesId
+            "entitiesId" to entitiesId,
+            "rights" to rights
         )
 
-        return session.query(query, parameters).map {
-            AvailableRightsForEntity(
-                it["id"] as String,
-                listOfNotNull(
-                    (it["right"] as Relationship?)?.type?.get(0),
-                    (it["grpRight"] as Relationship?)?.type?.get(0)
-                )
-            )
+        return session.query(query, parameters).filter {
```

Comment:
Do you still need the filter ?

Reply:
Now that you're talking about that, maybe this is not needed anymore, I need to do some checks to verify.

Reply:
I fixed the request to something better, more readable. Could you check ?

Reply:
It seems to be good.